.travis.yml

```
@@ -4,23 +4,24 @@ sudo: false
 jdk:
   - openjdk8
 env:
-  - GH_URL=https://raw.githubusercontent.com VALIDATOR_URL=http://localhost:8080/api/debug?url FILE_TO_VALIDATE=service-registry.yml URL_TO_VALIDATE=$GH_URL/${TRAVIS_PULL_REQUEST_SLUG:-$TRAVIS_REPO_SLUG}/${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}/$FILE_TO_VALIDATE
+  - GH_URL=https://raw.githubusercontent.com VALIDATOR_URL=http://localhost:8080/validator/debug?url FILE_TO_VALIDATE=service-registry.yaml URL_TO_VALIDATE=$GH_URL/${TRAVIS_PULL_REQUEST_SLUG:-$TRAVIS_REPO_SLUG}/${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}/$FILE_TO_VALIDATE
```

```suggestion
  - GH_URL=https://raw.githubusercontent.com VALIDATOR_URL=http://localhost:8080/validator/debug?url FILE_TO_VALIDATE=service-registry.yml URL_TO_VALIDATE=$GH_URL/${TRAVIS_PULL_REQUEST_SLUG:-$TRAVIS_REPO_SLUG}/${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}/$FILE_TO_VALIDATE
```

There is no such file `service-registry.yaml `

---

@mpanik I renamed the file as part of this PR (for consistency, we're using `.yaml` in other repositories) - am I missing something?

---

@mcupak you are right, my mistake

---

@mcupak anyway it seems the last two lines `VALIDATION_ERROR_COUNT` is to blame. We do already print output of validation. Is there any reason for having it as it is now in this PR? I do suggest simplifying it into:

before:
```
  # check if the validation output ignoring warnings is an empty array
  - VALIDATION_ERROR_COUNT=`echo "$VALIDATION_OUTPUT" |jq -c '.schemaValidationMessages | map(select(.level != "warning"))' |jq length`
  - test "$VALIDATION_ERROR_COUNT" == 0
```
after (not tested):
```
- if [ "$VALIDATION_OUTPUT" = "{}" ]; then VALIDATION_PASSED=true; else VALIDATION_PASSED=false; fi
- test "$VALIDATION_PASSED"==true
```

---

@mpanik I see what you mean, good catch! Your approach wouldn't work for OAS 3.0, because the tool currently throws validation warnings for unsupported features, so we need to filter them out. But you've correctly identified a case when if there are no warnings, `jq` returns null, and our count fails as a result. Let me fix that right now.

---

@mpanik How about now?

---

@mcupak LGTM